### ISSUE TITLE
CMake 3.28.1 Deprecation warning

### ISSUE THREAD

**Issue by FantasqueX** [libjpeg-turbo/libjpeg-turbo#740]
**Have you searched the existing issues (both open and closed) in the libjpeg-turbo issue tracker to ensure that this bug report is not a duplicate?**
Yes

**Does this bug report describe one of the [two known and unsolvable issues with the JPEG format](https://libjpeg-turbo.org/pmwiki/uploads/About/TwoIssueswiththeJPEGStandard.pdf)?**
No

**Clear and concise description of the bug:**
When I try to build libjpeg-turbo, there is a deprecation warning. My CMake version is 3.28.1.
```
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

According to CMake documentation https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html , this warning is added in CMake 3.27.

**Steps to reproduce the bug (using *only* libjpeg-turbo):**
```
mkdir build
cd build
cmake ..
```

**Image(s) needed in order to reproduce the bug (if applicable):**

**Expected behavior:**
No warning.

**Observed behavior:**
A deprecation warning.

**Platform(s) (compiler version, operating system version, CPU) on which the bug was observed:**
ArchLinux, CMake 3.28.1.

**libjpeg-turbo release(s), commit(s), or branch(es) in which the bug was observed (always test the tip of the main branch or the latest [stable pre-release](https://libjpeg-turbo.org/DeveloperInfo/PreReleases) to verify that the bug hasn't already been fixed):**
Current HEAD.

**If the bug is a regression, the specific commit that introduced the regression (use `git bisect` to determine this):**

**Additional information:**

**Comment by dcommander** [comment]
Some people rely on their operating system for mission-critical tasks and cannot adopt the most bleeding-edge version of it. Generally speaking, libjpeg-turbo has to support all of the enterprise/LTS distros that are still receiving public updates, and at least one of those has CMake 2.8.x. I'm sorry that the CMake developers chose to print an annoying warning regarding their deprecation decisions. That was a boneheaded move, IMHO, but it's not my fault or my problem. Our build system continues to work with CMake 2.8.x even if the CMake developers no longer support it.

**Comment by crisluengo** [comment]
You can fix this by changing the line 

    cmake_minimum_required(VERSION 2.8.12)

to

    cmake_minimum_required(VERSION 2.8.12...3.28)

It'll still work with 2.8.12.

**Comment by dcommander** [comment]
Well, great.  Apparently "fixing" this screwed up something in our Windows builds, so now all of the binaries depend on the Visual C++ runtime DLL.  (Normally, only **jpeg62.dll** should depend on it.)  Even more unfortunately, I didn't catch that before releasing 3.0.2, so now I have to figure out a fix and re-spin Windows binaries for 3.0.2.  I'm guessing it has something to do with a policy that is now set automatically because we claim to support up to CMake 3.28.  Ugh.  Seriously, no good deed ever goes unpunished with this project.
